src/Microsoft.DotNet.Build.Tasks/PackageFiles/executor/config.json

```diff
@@ -0,0 +1,126 @@
+﻿{
+  "properties": {
+    "OSGroup": {
+      "explanation": "Target OS for result binaries.",
+      "type": "property",
+      "values": "Windows_NT, Linux, OSX, FreeBSD, NetBSD",
+      "default": "Windows_NT"
+     },
+    "RestoreDuringBuild": {
+      "explanation": "Enables/Disables the restoration of packages.",
+      "type": "property",
+      "values": "True, False",
+      "default": true
+    },
+    "BuildTests": {
+      "explanation": "Enables/Disables building tests against the product binaries.",
+      "type": "property",
+      "values": "True, False",
+      "default": true
+    },
+    "BuildPackages": {
+      "explanation": "Enables/Disables building packages.", 
+      "type": "property",
+      "values": "True, False",
+      "default": true
+    },
+    "BuildTestsAgainstPackages": {
+      "explanation": "Allows to build tests against product packages.",
+      "type": "property",
+      "values": "True, False",
+      "default": false
+    },
+    "BatchRestorePackages": {
+      "explanation": "MsBuild target that restores the packages.",
+      "type": "target",
+      "values": "",
+      "default": ""
+    },
+    "BatchGenerateTestProjectJsons": {
+      "explanation": "MsBuild target that generates the project Jsons files to build tests against packages.",
+      "type": "target",
+      "values": "",
+      "default": ""
+    }
+  },
+  "commands": {
+    "simple": {
+      "explanation": "Only builds managed product binaries. It doesn't restore packages.",
+      "processName": "msbuild",
+      "project": "build.proj",
+      "extraCommands": "",
+      "property": {
+        "RestoreDuringBuild": false,
+        "BuildTests": false,
+        "BuildPackages": false
+      }
+    },
+    "complete": {
+      "explanation": "Restores packages, builds product and test binaries, runs the tests and builds product packages",
+      "processName": "msbuild",
+      "project": "build.proj",
+      "extraCommands": "",
+      "property": {
+        "RestoreDuringBuild": "",
+        "BuildTests": "",
+        "BuildPackages": ""
+      }
+    },
+    "testAgainstPackages": {
+      "explanation": "Generates project.jsons for test projects, restores packages, builds product and then builds tests against the generated project.json files.",
+      "processName": "msbuild",
+      "project": "build.proj",
+      "extraCommands": "",
+      "property": {
```

Comment:
As it is right now these are properties that are mandatory in order for the command to work.
If there is no value provided in the command or by the user, it would look for the default value previously defined.
The value should not be overwrite otherwise the command is not going to work.
Do we want more flexibility here?